packages/services/src/setting/index.ts

```diff
@@ -68,25 +68,34 @@ export class SettingManager extends DataConnector<
    *
    * @returns A promise that resolves if successful.
    */
-  async list(): Promise<{ ids: string[]; values: ISettingRegistry.IPlugin[] }> {
+  async list(
+    query?: 'ids'
+  ): Promise<{ ids: string[]; values: ISettingRegistry.IPlugin[] }> {
```

Comment:
Should we make `values` undefined in case only the ids are requested to avoid confusion with an hypothetical empty plugin list?

```suggestion
  ): Promise<{ ids: string[]; values?: ISettingRegistry.IPlugin[] }> {
```

Reply:
That would bring less confusion, I agree, but since the `SettingsManager` inherits from `DataConnector`, it should be also described in the `DataConnector` and its interface, I think.
That would change the API.